[Flaky Test] Test_diagError/Server_requires_client_auth

Test Name: Test_diagError/Server_requires_client_auth
Buildkite: https://buildkite.com/elastic/elastic-agent-libs/builds/632
Notes: It fails locally with stress


```
go run golang.org/x/tools/cmd/stress@latest ./httpcommon.test -test.run Test_diagError -test.v 

5s: 39 runs so far, 0 failures, 12 active

/var/folders/cv/_38x0gd965v0156jwblb9x6h0000gn/T/go-stress-20250306T134816-2830024646
=== RUN   Test_diagError
=== RUN   Test_diagError/TLS_server_no_client_CA
2025/03/06 13:48:25 http: TLS handshake error from 127.0.0.1:57959: read tcp 127.0.0.1:57958->127.0.0.1:57959: use of closed network connection
=== RUN   Test_diagError/HTTPS_schema_used_on_HTTP_server
=== RUN   Test_diagError/Server_cert_expired
2025/03/06 13:48:25 http: TLS handshake error from 127.0.0.1:57975: read tcp 127.0.0.1:57974->127.0.0.1:57975: use of closed network connection
=== RUN   Test_diagError/Server_requires_client_auth
2025/03/06 13:48:25 http: TLS handshake error from 127.0.0.1:57991: tls: client didn't provide a certificate
    diag_test.go:249: 
                Error Trace:    /Users/khushijain/Documents/elastic-agent-libs/transport/httpcommon/diag_test.go:249
                Error:          "Get \"https://127.0.0.1:57990\": readLoopPeekFailLocked: remote error: tls: certificate required" does not contain "caused by missing mTLS client cert."
                Test:           Test_diagError/Server_requires_client_auth
=== RUN   Test_diagError/Server_requires_client_auth,_client_cert_expired
2025/03/06 13:48:26 http: TLS handshake error from 127.0.0.1:58055: tls: failed to verify certificate: x509: certificate has expired or is not yet valid: current time 2025-03-06T13:48:26+05:30 is after 2025-03-04T19:18:26Z
--- FAIL: Test_diagError (1.04s)
    --- PASS: Test_diagError/TLS_server_no_client_CA (0.01s)
    --- PASS: Test_diagError/HTTPS_schema_used_on_HTTP_server (0.00s)
    --- PASS: Test_diagError/Server_cert_expired (0.16s)
    --- FAIL: Test_diagError/Server_requires_client_auth (0.22s)
    --- PASS: Test_diagError/Server_requires_client_auth,_client_cert_expired (0.66s)
FAIL


ERROR: exit status 1

10s: 87 runs so far, 1 failures (1.15%), 12 active
15s: 127 runs so far, 1 failures (0.79%), 12 active
^Csignal: interrupt
```

--- a/transport/httpcommon/diag.go
+++ b/transport/httpcommon/diag.go
@@ -192,7 +192,7 @@ func diagError(err error) string {
 			return fmt.Sprintf("%v:  caused by missing mTLS client cert.", uErr)
 		}
 		if strings.Contains(errString, "remote error: tls: expired certificate") {
-			return fmt.Sprintf("%v: caused by expired mTLS client cert..", uErr)
+			return fmt.Sprintf("%v: caused by expired mTLS client cert.", uErr)
 		}
 		if strings.Contains(errString, "remote error: tls: bad certificate") {
 			return fmt.Sprintf("%v: caused by invalid mTLS client cert, does the server trust the CA used for the client cert?.", uErr)

--- a/transport/httpcommon/diag.go
+++ b/transport/httpcommon/diag.go
@@ -30,6 +30,7 @@ import (
 	"net/http/httptrace"
 	"net/textproto"
 	"net/url"
+	"strings"
 
 	"github.com/elastic/elastic-agent-libs/transport/tlscommon"
 )
@@ -181,15 +182,19 @@ func diagError(err error) string {
 	// keep unwrapping to url.Error as the last error as other failures can be embedded in a url.Error
 	// Can detect if an HTTPS request is made to an HTTP server
 	var uErr *url.Error
+
 	if errors.As(err, &uErr) {
-		switch uErr.Err.Error() {
-		case "http: server gave HTTP response to HTTPS client":
+		errString := uErr.Err.Error()
+		if strings.Contains(errString, "http: server gave HTTP response to HTTPS client") {
 			return fmt.Sprintf("%v: caused by using HTTPS schema on HTTP server.", uErr)
-		case "remote error: tls: certificate required":
-			return fmt.Sprintf("%v: caused by missing mTLS client cert.", uErr)
-		case "remote error: tls: expired certificate":
-			return fmt.Sprintf("%v: caused by expired mTLS client cert.", uErr)
-		case "remote error: tls: bad certificate":
+		}
+		if strings.Contains(errString, "remote error: tls: certificate required") {
+			return fmt.Sprintf("%v:  caused by missing mTLS client cert.", uErr)
+		}
+		if strings.Contains(errString, "remote error: tls: expired certificate") {
+			return fmt.Sprintf("%v: caused by expired mTLS client cert..", uErr)
+		}
+		if strings.Contains(errString, "remote error: tls: bad certificate") {
 			return fmt.Sprintf("%v: caused by invalid mTLS client cert, does the server trust the CA used for the client cert?.", uErr)
 		}
 	}

